### PR TITLE
Add more documentation and examples for MockDateTime

### PR DESCRIPTION
**Pull request by gregtatum**
I know this utility is mostly for tests, but I was [working on understanding some date/time functionality](https://github.com/gregtatum/spec-rs/blob/e05183ac5fa895d6ee34263de8f9ae4f36101d87/src/icu_test.rs#L152-L154) and documented my learnings on the way.

### CODE REVIEW

File: components/datetime/src/date.rs [hunk 1]
```diff
@@ -138,6 +147,17 @@ impl DateTimeType for MockDateTime {
 impl FromStr for MockDateTime {
     type Err = DateTimeError;
 
+    /// Parse a `MockDateTime` from a string.
+    ///
+    /// This utility is for easily creating dates, not a complete robust solution. The
+    /// string must take a specific form of the ISO 8601 format: `YYYY-MM-DDThh:mm:ss`.
```
Comment (gregtatum) [inline components/datetime/src/date.rs:153]: Is this correct?

Comment (zbraniecki) [reply components/datetime/src/date.rs:153]: yes!

File: components/datetime/src/date.rs [hunk 2]
```diff
@@ -156,6 +176,9 @@ impl FromStr for MockDateTime {
     }
 }
 
+/// This macro defines a struct for each type of unit to be used in a DateTime. Each
+/// unit is bounded by a range. The traits implemented here will return a Result on
+/// whether or not the unit is in range from the given input.
```
Comment (zbraniecki) [inline components/datetime/src/date.rs:181]: The line breaks here are weird. Can you realign?

Comment (gregtatum) [reply components/datetime/src/date.rs:181]: Can you clarify what the line-break style should look like? I was struggling to understand it. In other projects I've done hard cut offs at a column width, so things flow where they happen to end. I was noticing that a lot of the docs have very long line widths. What is the cut-off strategy?

File: components/datetime/src/date.rs [hunk 3]
```diff
@@ -48,14 +48,22 @@ pub trait DateTimeType: FromStr {
 /// Temporary implementation of [`DateTimeType`],
 /// which is used in tests, benchmarks and examples of this component.
 ///
+/// *Notice:* Rust at the moment does not have a canonical way to represent date and time. We are introducing
+/// `MockDateTime` as an example of the data necessary for ICU [`DateTimeFormat`] to work, and
+/// we hope to work with the community to develop core date and time APIs that will work as an input for this component.
```
Comment (sffc) [inline components/datetime/src/date.rs:53]: Can you add a link to https://github.com/unicode-org/icu4x/blob/master/docs/research/date_time.md ?

File: components/datetime/src/date.rs (hunk 2 above)
Comment (zbraniecki) [reply components/datetime/src/date.rs:181]: Ah, no, retracting. I'm not sure why Github showed me this comment malformed. It looks good now.

File: components/datetime/src/date.rs (hunk 3 above)
Comment (gregtatum) [reply components/datetime/src/date.rs]: Sure, that helps with the clarity here. I also added links to the other locations with similar text.

**Pull request merged**